lib/channel.js

```diff
@@ -136,29 +136,35 @@ class Channel {
     try {
       result = await proc.invoke(context, args);
     } catch (error) {
-      const timedout = error.message === 'Timeout reached';
-      const code = timedout ? 408 : 500;
-      this.error(code, { callId, error, pass: timedout });
+      const isTimeout = error.message === 'Timeout reached';
+      const httpCode = error.httpCode || (isTimeout ? 408 : 500);
+      this.error(error.code, { callId, error, httpCode });
       return;
     } finally {
       proc.leave();
     }
-    if (typeof result === 'object' && result.constructor.name === 'Error') {
-      this.error(result.code, { callId, error: result, pass: true });
+    if (
+      result !== null &&
```

Comment:
We do not need this check, because `null` will be covered by `typeof result` check

Reply:
Can you clarify, `typeof null === 'object'`?

Reply:
My mistake, I think we need `if (result && result.constructor && result.constructor.name === 'Error') {`